app/main.py

```diff
@@ -1,2 +1,118 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.length_hash = 8
+        self.length_dict = 0
+        self.capacity = [[] for _ in range(self.length_hash)]
+
+    def __setitem__(self, key, value):
+        if self.length_dict >= self.length_hash * 2 // 3:
+            self.__resize_hash()
+        hash_ = hash(key)
+        new_index = hash_ % self.length_hash
+        hash_value = new_index
+        while True:
+            if self.capacity[new_index]:
+                if hash_value == self.capacity[new_index][1] and\
+                        self.capacity[new_index][0] == key:
+                    if self.capacity[new_index][2] != value:
+                        self.capacity[new_index][2] = value
+                        return
+                elif new_index == self.length_hash - 1:
+                    new_index = 0
+                else:
+                    new_index += 1
+            else:
+                self.capacity[new_index] = [key, hash_, value]
+                self.length_dict += 1
+                return
+
+    def __getitem__(self, key):
+        hash_ = hash(key)
+        new_index = hash_ % self.length_hash
+        count = 0
+        while True:
+            if not self.capacity[new_index]:
+                raise KeyError(f"{key} is not founded")
+            if self.capacity[new_index] and\
+                    self.capacity[new_index][0] == key and\
+                    self.capacity[new_index][1] == hash_:
+                return self.capacity[new_index][2]
+            elif new_index == self.length_hash - 1:
+                new_index = 0
+            else:
+                new_index += 1
+            count += 1
+
+    def __len__(self):
+        return self.length_dict
+
+    def __delitem__(self, key):
+        hash_ = hash(key)
+        hash_index = hash_ % len(self.capacity)
+        while True:
+            if not self.capacity[hash_index]:
+                raise KeyError(f"{key} is not founded")
+            if self.capacity[hash_index] and \
+                    self.capacity[hash_index][1] == hash_ and\
+                    self.capacity[hash_index][0] == key:
+                pop_item = self.capacity[hash_index]
+                self.capacity[hash_index].clear()
```

Comment:
You cannot just clear the corresponding cell

Reply:
<img width="777" alt="image" src="https://user-images.githubusercontent.com/44612098/191455360-750990d7-f6af-4440-8d90-5f9f5d04570b.png">

Reply:
Comment where I describe this issue